codegen/funcs.go

```diff
@@ -155,3 +156,59 @@ func KebabCase(name string) string {
 	name = SnakeCase(name)
 	return strings.Replace(name, "_", "-", -1)
 }
+
+// GoTypeRef returns the Go code that refers to the Go type which matches the given data type
+func GoTypeRef(dt design.DataType) string {
+	tname := GoTypeName(dt)
+	if design.IsObject(dt) {
+		return "*" + tname
+	}
+	return tname
+}
+
+// GoTypeName returns the Go type name for a data type.
+func GoTypeName(dt design.DataType) string {
+	switch actual := dt.(type) {
+	case design.Primitive:
+		return GoNativeType(dt)
+	case *design.Array:
+		return "[]" + GoTypeRef(actual.ElemType.Type)
```

Comment:
Please add a TBD comment that other types need to be added here (map, object, user type, etc.)